UKHomeOffice/sgmr-service
# SPCR-386 Create a conditional pleasure craft component

### AC / Description of changes
Update the pleasure craft (PC) selection in the voyage flow depending on the number of PCs associated with a user.
Create a pleasure craft component
If the user has 0 PCs - render select PC page (with no PCs in list) with add PC option
If the user has more then one PC - render select PC page with list of PCs and the add PC option
If the user only has 1 PC - render a "Will you sail in x?" page (if user clicks YES - select people page. If NO - add PC form)
Add validation
Link to new pleasure craft form

### To Test
- Delete all saved pleasure crafts
- Click on 'Create a new voyage plan'
- Complete the departure & arrival form and continue
- You should see a page asking you to add a pleasure craft
- Click on Save and **continue**
- >There should be an error reading 'You must select a pleasure craft'
- Click on 'Add a new pleasure craft'
- Click on Save and **continue**
- > You should be taken to the pleasure craft form
- Enter pleasure craft details
- > You should be taken back to the voyage flow
- > The page should ask if you are sailing in the PC you just saved
- Click save and **continue**
- >There should be an error reading 'You must select whether you are sailing in this pleasure craft'
- Click yes
- Click save and **continue**
- >You should be taken to the people page
- Go back and click no
- Click save and **continue**
- >You should be taken to the pleasure craft form (**Note:** There is a bug here where details will already be filled in - a ticket is being made to fix this)
- Add a new pleasure craft
- > You should be taken back to the voyage flow
- > The page should display a list with the 2 PCs you saved and an option to add a new PC
- Select a pleasure craft 
- Click save and **exit**
- > You should be taken back to the dashboard
- Go to manage reports and look at your drafts
- > You should see the voyage you created with the correct PC listed
### Notes

Along with the form pre-filling bug above - there is another bug when, if there are no PCs saved, you can see the page asking you to add a PC. If you click on save and continue without selecting anything and then click on the error link - it does not work as currently it is coded to work for the list of PCs

---
* remember to merge to feature branches not master *
